### DISCUSSION TITLE
App not rendering on linux

### DISCUSSION THREAD

**Discussion by joelphilippage** [AvaloniaUI/Avalonia#14569] [Q&A] [answered]
Hello,
I am creating my first app with Avalonia. I want to use it as a container for my web game. The app is very basic, it just contains a web view from this library and runs a site:
https://github.com/MicroSugarDeveloperOrg/Avalonia.WebView

The app builds and runs fine on Windows and Mac, but I'm trying to run it on a steam deck which runs Arch Linux. It seems to launch, but never renders anything.
Here's a screenshot. The console hangs up and I can see it running in the task manager but nothing happens.
![image](https://github.com/AvaloniaUI/Avalonia/assets/13322475/216beddd-409f-4b76-9861-0bfab43dbea2)
Is there any way to see logs? It doesn't seems to crash.
I also tried on the latest version of Ubuntu and had the exact same result.
I use this to build for linux
`dotnet publish --runtime linux-x64 --self-contained`
The linux build is 64 bit.

Please let me know if there is any other information I can provide that would be helpful. Thank you!

**Comment by thevortexcloud** [comment]
> but I'm trying to run it on a steam deck which runs Arch Linux.

Are you running actual Arch or SteamOS? They are very different things. SteamOS is significantly more restricted than normal Arch and is typically very far behind Arch in terms of updates. Ubuntu is also typically very far behind in updates (sometimes >2 years).

In any case have you tested running the default template? I personally have had no issues with getting an Avalonia window to appear on Linux. But I have also not tried the web view control. It's possible you are missing a native library it needs. Looking at the code it looks like it has dependencies on GTK and Webkit on Linux. Neither of which may be installed depending on your distro. Even if you do have the library it  may be called something different. Consistent naming of libraries between distros is not a given.

> I am creating my first app with Avalonia. I want to use it as a container for my web game.

This seems like an odd thing to do? Do you specifically need Avalonia for this? I can't believe I am saying this: but it sounds like you actually want something like Electron.

**Comment by joelphilippage** [comment, marked as answer]
Thank you! I haven't done much with Linux so didn't think about another library having extra dependencies. I had to install webkit2gtk to get it to run. I guess I expected an error or something. I will check out Electron to see if it will work better for Linux then Avalonia. I was hoping to add a built-in editor and I'm more used to programming with WinUI and C#

**Reply by thevortexcloud** [reply]
Glad I could help!

>I haven't done much with Linux so didn't think about another library having extra dependencies

You would have also run onto this problem with Windows if what ever native library it needs was missing. It's just that distributing native libraries on Linux is fairly complicated as they typically need to be built against the distro they are running on (or a fairly old distro that most other moden distro is compatible with). Otherwise they may into problems like libc incompatibilities. Which is also why things like Flatpak have taken off in recent years since it side steps problems like that entirely.

>  I guess I expected an error or something. 

.NET will typically give errors if it can't find a native library, but they may be getting swallowed somewhere. A lot of people do try catches and then just do nothing with the error. 

>  I will check out Electron to see if it will work better for Linux then Avalonia.

They both work fine (although I am not a fan of Electron due to how heavy it is), Electron specifically though is used by most major cross platform apps that run on Linux, EG Teams, Discord, etc. It just seemed odd to me to go through the effort of wrapping a web browser to only display a web app instead of using what is functionally a web browser. If you are planning to add additional functionality besides your game, then that makes far more sense.

**Reply by rabbitism** [reply]
this is explained in readme.

use Webkit Webview for Linux (if not please install [sudo apt-get libwebkit2gtk-4-37])

**Reply by thevortexcloud** [reply]
Those are Ubuntu/Debian instructions. They wont work on all distros, and certainly wont work on SteamOS.
